### PR TITLE
nifm/ssl: Implement GetCurrentNetworkProfile and stub Ssl Service

### LINKED ISSUE #643: nifm:u IGeneralService: 5 (GetCurrentNetworkProfile) not implemented
IGeneralService GetCurrentNetworkProfile is not implemented.

## Service description:
[Switchbrew](https://switchbrew.org/wiki/Network_Interface_services#IGeneralService) | [Reswitched](https://reswitched.github.io/SwIPC/ifaces.html#nn::nifm::detail::IGeneralService(5))

```
GetCurrentNetworkProfile() -> buffer<nn::nifm::detail::sf::NetworkProfileData, 0x1a, 0x17c>
```

Unknown.

## Required by:
- https://github.com/Ryujinx/Ryujinx-Games-List/issues/47

### LINKED ISSUE #1289: ssl ISslContext: 5 (ImportClientPki) is not implemented
ssl ISslContext: 5 (ImportClientPki) is not implemented

[Switchbrew](https://switchbrew.org/wiki/SSL_services#ImportClientPki) | [SwIPC](https://reswitched.github.io/SwIPC/ifaces.html#nn::ssl::sf::ISslContext(5))

## Service description:
```
Takes two type-0x5 input buffers, returns an output u64 Id.
The first buffer contains the PKCS#12 data, the second buffer contains the optional (addr=NULL/size=0) ASCII password. 
The password is copied to a heap buffer with size+1, for NUL-termination.
An error is thrown if this cmd or #RegisterInternalPki was already used previously.
```

## Required by:
https://github.com/Ryujinx/Ryujinx-Games-List/issues/1519
https://github.com/Ryujinx/Ryujinx-Games-List/issues/2798

### PR DESCRIPTION
**Pull request by AcK77**
This PR implement the `nifm` call GetCurrentNetworkProfile (REd by @Thog, thanks to her) and stubs some `ssl` services calls, some details:

- `nifm` `IGeneralService` `GetCurrentNetworkProfile` is implemented (fixes #643) and now blocked games boot further.
- `nifm` `GetLocalInterface()` helper and `DnsSetting` struct have been fixed to return the right values.
- `ssl` `ISslService` have been cleaned up.
- `ssl` `ISslContext` have been cleaned up, `CreateConnection` and `ImportClientPki` have been stubbed (fixes #1289).
- `ssl` `ISslConnection` is partially stubbed, which make some games playable, see the attached screenshots.
- `nsd` `IManager` `Resolve` and `ResolveEx` have been fixed using the right buffer type.

All those changes makes some games boot further and some others playable:

image
image
image
image
image
image
image
image
image
image
image
image

### CODE REVIEW

File: Ryujinx.HLE/HOS/Services/Sockets/Nsd/IManager.cs [hunk 1]
```diff
@@ -132,7 +132,8 @@ public ResultCode Unknown(ServiceCtx context)
         // Resolve(buffer<unknown<0x100>, 0x15>) -> buffer<unknown<0x100>, 0x16>
         public ResultCode Resolve(ServiceCtx context)
         {
-            (long outputPosition, long outputSize) = context.Request.GetBufferType0x22();
+            long outputPosition = context.Request.ReceiveBuff[0].Position;
```
Comment (Thog) [inline Ryujinx.HLE/HOS/Services/Sockets/Nsd/IManager.cs:135]: Nice catch 👍

File: Ryujinx.HLE/HOS/Services/Sockets/Nsd/IManager.cs [hunk 2]
```diff
@@ -147,7 +148,8 @@ public ResultCode Resolve(ServiceCtx context)
         // ResolveEx(buffer<unknown<0x100>, 0x15>) -> (u32, buffer<unknown<0x100>, 0x16>)
         public ResultCode ResolveEx(ServiceCtx context)
         {
-            (long outputPosition, long outputSize) = context.Request.GetBufferType0x22();
+            long outputPosition = context.Request.ReceiveBuff[0].Position;
```
Comment (Thog) [inline Ryujinx.HLE/HOS/Services/Sockets/Nsd/IManager.cs:151]: Nice catch 👍

File: Ryujinx.HLE/HOS/Services/Ssl/Types/InterfaceVersion.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Ryujinx.HLE.HOS.Services.Ssl.Types
+{
+    enum InterfaceVersion : uint
```
Comment (Thog) [inline Ryujinx.HLE/HOS/Services/Ssl/Types/InterfaceVersion.cs:3]: This is actually just a version number to me

**Review by Thog — APPROVED** [review]
lgtm only one thing

**Review by riperiperi — APPROVED** [review]
Looking good, I eagerly await doing the LDN rebase 🥴

**Pull request merged**